File: check_imports.py
import os, glob

# Get all files in lib/services
services_dir = r'c:\Users\alice\medibuddy-Project\lib\services'
service_files = [f for f in os.listdir(services_dir) if f.endswith('.dart')]

# Get all dart files in lib/
all_dart_files = []
for root, dirs, files in os.walk(r'c:\Users\alice\medibuddy-Project\lib'):
    for file in files:
        if file.endswith('.dart'):
            all_dart_files.append(os.path.join(root, file))

# Count imports
usage = {f: [] for f in service_files}

for dart_file in all_dart_files:
    try:
        with open(dart_file, 'r', encoding='utf-8') as f:
            content = f.read()
            for sf in service_files:
                if os.path.basename(dart_file) != sf:
                    # check if the word sf is in content
                    if sf in content:
                        usage[sf].append(dart_file)
    except Exception as e:
        pass

print('--- Usage Report ---')
for sf in service_files:
    print(f'{sf}: {len(usage[sf])} references')
    for ref in usage[sf]:
        print(f'  - {ref}')

File: check_imports.dart
import 'dart:io';

void main() async {
  final servicesDir =
      Directory('c:/Users/alice/medibuddy-Project/lib/services');
  final libDir = Directory('c:/Users/alice/medibuddy-Project/lib');

  final serviceFiles = servicesDir
      .listSync()
      .whereType<File>()
      .where((f) => f.path.endsWith('.dart'))
      .map((f) => f.uri.pathSegments.last)
      .toList();

  final allDartFiles = libDir
      .listSync(recursive: true)
      .whereType<File>()
      .where((f) => f.path.endsWith('.dart'))
      .toList();

  final usage = <String, List<String>>{};
  for (final sf in serviceFiles) {
    usage[sf] = [];
  }

  for (final file in allDartFiles) {
    try {
      final content = file.readAsStringSync();
      final currentFile = file.uri.pathSegments.last;

      for (final sf in serviceFiles) {
        if (currentFile != sf) {
          if (content.contains(sf)) {
            usage[sf]!.add(file.path);
          }
        }
      }
    } catch (e) {
      // ignore
    }
  }

  print('--- Usage Report Custom ---');
  for (final sf in serviceFiles) {
    print('\$sf: \${usage[sf]!.length} references');
    if (usage[sf]!.isEmpty) {
      print('  -> NOT USED');
    }
    for (final ref in usage[sf]!) {
      print('  - \$ref');
    }
  }
}
